### PR TITLE
TELCODOCS-1879 - Adding docs for PTP Operator T-GM leap seconds feature

### PR DESCRIPTION
**Pull request by aireilly**
Adding docs for PTP Operator T-GM leap seconds feature delivered in CNF-12603.

Also removes duplicate `snippets/ptp_PtpConfigDualCardGmWpc.yaml` and `snippets/ptp_PtpConfigGmWpc.yaml` files which were added in error.

Version(s):
4.15+
hold for 4.14.z 

Issue:
https://issues.redhat.com/browse/TELCODOCS-1879

Link to docs preview:
* https://79493--ocpdocs-pr.netlify.app/openshift-enterprise/latest/networking/ptp/about-ptp.html#ptp-overview-of-gnss-grandmaster-clock_about-ptp
* https://79493--ocpdocs-pr.netlify.app/openshift-enterprise/latest/networking/ptp/configuring-ptp.html#ptp-configuring-dynamic-leap-seconds-handling-for-tgm_configuring-ptp

QE review:
- [ ] QE has approved this change.

### CODE REVIEW

**Comment by aireilly** [comment]
Closing to kick Prow

File: modules/ptp-configuring-dynamic-leap-seconds-handling-for-tgm.adoc [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+[id="ptp-configuring-dynamic-leap-seconds-handling-for-tgm_{context}"]
+= Configuring dynamic leap seconds handling for PTP grandmaster clocks
+
+The PTP Operator container image includes the latest `leap-seconds.list` file  that is available at the time of release.
+You can configure the PTP Operator to automatically update the leap second file by using GPS announcements.
+
+Leap second information is stored in an automatically generated `leap-configmap` custom resource (CR) in the `openshift-ptp` namespace.
+The PTP Operator mounts the `leap-configmap` CR as a volume in the `linuxptp-daemon` pod that is accessible by the `ts2phc` process.
+
+If the GPS satellite broadcasts new leap second data, the PTP Operator updates the `leap-configmap` CR with the new data.
```
Comment (vitus133) [inline modules/ptp-configuring-dynamic-leap-seconds-handling-for-tgm.adoc:15]: Technically ConfigMap is not a custom resource (CR). Custom resources are only those defined by a CustomResourceDefinition (CRD). But ConfigMap is a **core** resource.

```suggestion
Leap second information is stored in an automatically generated ConfigMap resource named `leap-configmap` in the `openshift-ptp` namespace.
The PTP Operator mounts the `leap-configmap` as a volume in the `linuxptp-daemon` pod that is accessible by the `ts2phc` process.

If the GPS satellite broadcasts new leap second data, the PTP Operator updates the `leap-configmap` with the new data.
```

File: modules/ptp-configuring-dynamic-leap-seconds-handling-for-tgm.adoc [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+<1> Set `-w` to force `phc2sys` to wait until `ptp4l` has synchronized the system hardware clock before starting its own synchronization process.
++
+[NOTE]
+====
+Previously, you applied an offset adjustment in the `phc2sys` configuration (`-O -37`) to account for historical leap seconds.
+This is no longer required.
+====
+
+. Configure the Intel e810 NIC to
+enable reading of `NAV-TIMELS` messages from the GPS receiver in the `spec.profile.plugins.e810.ublxCmds` section of the `PtpConfig` CR.
```
Comment (vitus133) [inline modules/ptp-configuring-dynamic-leap-seconds-handling-for-tgm.adoc:50]: The software is not reading the information, it is configuring the GPS module to report it
```suggestion
enable reporting of `NAV-TIMELS` messages by the GPS receiver in the `spec.profile.plugins.e810.ublxCmds` section of the `PtpConfig` CR.
```

File: modules/ptp-configuring-dynamic-leap-seconds-handling-for-tgm.adoc [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+For example:
++
+[source,yaml]
+----
+- args: #ubxtool -P 29.20 -p CFG-MSG,1,38,300
+    - "-P"
+    - "29.20"
+    - "-p"
+    - "CFG-MSG,1,38,300"
+----
```
Comment (vitus133) [inline modules/ptp-configuring-dynamic-leap-seconds-handling-for-tgm.adoc:60]: 300 is an out-of-bounds number, there was a bug that we fixed...
```suggestion
----
- args: #ubxtool -P 29.20 -p CFG-MSG,1,38,248
    - "-P"
    - "29.20"
    - "-p"
    - "CFG-MSG,1,38,248"
----
```

File: modules/ptp-configuring-dynamic-leap-seconds-handling-for-tgm.adoc [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+.Verification
+
+You can validate that the leap-second `ConfigMap` CR has been successfully generated by the PTP Operator.
```
Comment (vitus133) [inline modules/ptp-configuring-dynamic-leap-seconds-handling-for-tgm.adoc:64]:
```suggestion
You can validate that the leap-configmap `ConfigMap` has been successfully generated by the PTP Operator.
```

File: modules/ptp-configuring-dynamic-leap-seconds-handling-for-tgm.adoc [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+
+Following a leap event announcement sent by GPS, you can check that the PTP Operator changed the local copy of the leap seconds file.
```
Comment (vitus133) [inline modules/ptp-configuring-dynamic-leap-seconds-handling-for-tgm.adoc:66]: It's simple to inspect the configmap, but to inspect the local copy of the leap seconds file the user will need to rsh into the pod, check where the leap seconds file is read from by ts2phc and  inspect it. I just anticipate the question "how do you want us to check the leap seconds file"...
```suggestion
Following a leap event announcement sent by GPS, you can check that the PTP Operator changed the local copy of the leap seconds information by inspecting the leap-configmap.
```

**Review by vitus133 — COMMENTED** [review]
I shared a couple of thoughts. Please feel free to reject the suggestions and rephrase it as needed.
Thanks!

**Review by michaelryanpeter — COMMENTED** [review]
I left a few things to consider. Crisp prose, as always.

/label peer-review-done
/remove-label peer-review-in-progress
/remove-label peer-review-needed

File: modules/ptp-configuring-dynamic-leap-seconds-handling-for-tgm.adoc [hunk 6]
```diff
@@ -0,0 +1,153 @@
+// Module included in the following assemblies:
+//
+// * networking/ptp/configuring-ptp.adoc
+
+:_mod-docs-content-type: PROCEDURE
+[id="ptp-configuring-dynamic-leap-seconds-handling-for-tgm_{context}"]
+= Configuring dynamic leap seconds handling for PTP grandmaster clocks
+
+The PTP Operator container image includes the latest `leap-seconds.list` file  that is available at the time of release.
```
[the remark this answers is not in the document]
Comment (aireilly) [reply modules/ptp-configuring-dynamic-leap-seconds-handling-for-tgm.adoc:9]: Thanks @michaelryanpeter I made some updates to expand PTP in various places. The operator name is "PTP Operator" so I am not going to expand the abbreviation in the Operator name.

**Comment by aireilly** [comment]
Waiting on z stream release to CP to enterprise-4.14.

File: modules/ptp-configuring-dynamic-leap-seconds-handling-for-tgm.adoc (hunk 6 above)
[the remark this answers is not in the document]
Comment (michaelryanpeter) [reply modules/ptp-configuring-dynamic-leap-seconds-handling-for-tgm.adoc:9]: SGTM. Thanks for taking the team. Though the changes you made are subtle, I do find it a easier to read as someone unfamiliar with the topic.

**Pull request merged**